projects/storefrontlib/layout/loading/intersection.service.ts

```diff
@@ -91,6 +95,20 @@ export class IntersectionService {
     element: HTMLElement,
     options: IntersectionOptions
   ): Observable<IntersectionObserverEntry[]> {
+    if (isPlatformServer(this.platformId)) {
+      return of([
+        {
+          target: element,
+          isIntersecting: true,
+          intersectionRatio: 1,
+          boundingClientRect: {} as DOMRectReadOnly,
+          intersectionRect: {} as DOMRectReadOnly,
+          rootBounds: {} as DOMRectReadOnly,
+          time: Date.now(),
+        } as IntersectionObserverEntry,
+      ]);
+    }
```

Comment:
`major:` in my opinion it is rather healing the symptoms instead of fixing the root cause. Instead of mocking the API not available in the Node.js, let's adjust the logic instead to not run/create intersection observer on a server side (it won't work anyway)